src/main/java/frc/team2412/robot/subsystems/LimelightSubsystem.java

```diff
@@ -0,0 +1,196 @@
+				.withPosition(0, 2)
+				.withSize(4, 1);
+	}
+
+	private void setPipeline() {
+		networkTable.getEntry("pipeline").setNumber(2);
+	}
+
+	// METHODS
+
+	public boolean hasTargets() {
+		return (networkTable.getEntry("tv").getDouble(0) != 0);
+	}
+
+	public double getHorizontalOffset() {
+		return networkTable.getEntry("tx").getDouble(0);
+	}
+
+	public double getVerticalOffset() {
+		return networkTable.getEntry("ty").getDouble(0);
+	}
+
+	public double getBoxWidth() {
+		return networkTable.getEntry("tshort").getDouble(0);
+	}
+	//TODO re-measure distances to make this right
+	public double getDistanceFromTarget() {
+
+		// focal length = (P x D) / W
+		double focal_length = 346.1818;
+
+		// distance = (W x F) / P
+		// returns inches for testing purposes, will divide by 39.3700787 to return meters
+		return (8.25 * focal_length) / getBoxWidth();
+	}
+
+	// returns turn power from exponential curve, input is from -27deg to +27deg
+	public double turnPowerExp() {
+		if (Math.abs(getHorizontalOffset()) >= 1.0) {
+			return MathUtil.clamp(
+					(Math.signum(getHorizontalOffset())
+							* (Math.pow(Math.abs(getHorizontalOffset()), 0.6) * 0.14)),
+					-0.25,
+					0.25);
+		} else {
+			return 0.0;
+		}
+	}
+
+	// returns turn power from linear curve, input is from -27deg to +27deg
+	public double turnPowerLin() {
+		if (Math.abs(getHorizontalOffset()) >= 1.0) {
+			return MathUtil.clamp(0.04 * getHorizontalOffset(), -0.25, 0.25);
+		} else {
+			return 0.0;
+		}
+	}
+
+	// tan(degree) * distance = sideways distance
+
+	// target height / tan(vertical angle)
+
+	public Pose2d getTargetPose(Pose2d currentPose) {
+
+		// math thing to get target pose using current pose
+
+		Rotation2d targetHeading =
+				new Rotation2d(
+						currentPose.getRotation().getRadians() + Units.degreesToRadians(getHorizontalOffset()));
+		double targetDistance = getDistanceFromTarget() / 39.3700787;
+
+		double targetX = Math.sin(targetHeading.getRadians()) * targetDistance;
+		double targetY = Math.cos(targetHeading.getRadians()) * targetDistance;
+
+		Pose2d targetPose =
+				new Pose2d(currentPose.getX() + targetY, currentPose.getY() + targetX, targetHeading);
+
+		currentPoseString = currentPose.toString();
+		targetPoseString = targetPose.toString();
+
+		return targetPose;
+	}
+
+	public String getCurrentPoseString() {
+		return currentPoseString;
+	}
+
+	public String getTargetPoseString() {
+		return targetPoseString;
+	}
+
+	public boolean isWithinDistance() {
+		return (getDistanceFromTarget() <= GOAL_DISTANCE_FROM_TARGET);
+	}
+
+	public Command getWithinDistance(DrivebaseSubsystem drivebaseSubsystem) {
+		final DoubleSupplier returnTurn = () -> turnPowerLin();
+		final DoubleSupplier returnZero = () -> 0.0;
+		Command moveCommand;
+		if (hasTargets()) {
+			System.out.println("has targets");
+			moveCommand =
+					new DriveCommand(drivebaseSubsystem, returnZero, returnZero, returnTurn, returnZero);
```

Comment:
Why are you using DriveCommand instead of DrivebaseSubsystem.rotateToAngle?

Reply:
mostly because the end goal is to have it rotate and drive at the same time for better efficiency. Driving is just turned off not to cause osha violations

Reply:
rotate and drive as in drive to the note? or rotate and drive as in rotate as the driver is driving? If you're trying to go for the first option, PathPlanner has functions that can generate paths like that for you. If you're going for the second option, rotateToAngle is built to handle that. Either way, adding a whole new DriveCommand (which will get confused with DrivebaseSubsystem's driveJoystick) seems like a very complex way to solve this issue

Reply:
Rotate and drive to the note simultaneously. Kirby and I tried to get pathplanner to generate paths in that way and it never worked. Either way, drivecommand will probably be changed in favor of just using either drive() or drivejoystick().

Reply:
The new DriveToNoteCommand() works and uses the drive() method